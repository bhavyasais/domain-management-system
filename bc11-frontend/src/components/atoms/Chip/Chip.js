import React from "react";
import MuiChip from "@material-ui/core/Chip";
import { withStyles } from "@material-ui/core";
import PropTypes from "prop-types";
import clsx from "clsx";

const styles = (theme) => ({
  chip: {
    maxWidth: "150px",
    fontFamily: "Roboto",
  },
});

const Chip = ({
  label,
  backgroundColor,
  color,
  size,
  chipStyle,
  variant,
  classes,
  ...restProps
}) => {
  return (
    <MuiChip
      label={label}
      style={{ backgroundColor: backgroundColor, color: color }}
      size={size}
      className={clsx(classes.chip, chipStyle)}
      variant={variant}
      {...restProps}
    />
  );
};

Chip.propTypes = {
  label: PropTypes.string.isRequired,
  color: PropTypes.string,
  size: PropTypes.string,
  variant: PropTypes.string,
  backgroundColor: PropTypes.string,
};
Chip.defaultProps = {
  backgroundColor: "#eeeeee",
  color: "#1c1c21",
  size: "small",
  chipStyle: null,
  variant: "default",
};

export default withStyles(styles)(Chip);
