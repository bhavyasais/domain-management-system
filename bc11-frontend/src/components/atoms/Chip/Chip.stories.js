import React from "react";
import Chip from "./Chip";
import { withKnobs, text, select, color } from "@storybook/addon-knobs";

export default {
  component: Chip,
  title: "Atoms/Chip",
  decorators: [withKnobs],
};

export const defaultChip = ({ classes }) => (
  <Chip
    label={text("label", "1000")}
    backgroundColor={color("backgroundColor", "#ffada5")}
    color={color("color", "black")}
    variant={select("variant", ["default", "outlined"], "default")}
  />
);
