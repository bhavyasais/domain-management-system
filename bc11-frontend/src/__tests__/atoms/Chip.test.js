import Chip from "../../components/atoms/Chip/Chip";
import React from "react";
import renderer from "react-test-renderer";
import { mount } from "enzyme";
import { act } from "react-dom/test-utils";

describe("<Chip/>", () => {
  let mounted;
  beforeEach(() => {
    act(() => {
      mounted = mount(<Chip label="Test label" />);
    });
  });

  it("Renders correctly", () => {
    let wrapper;
    wrapper = renderer.create(<Chip label="Test label" />);
    expect(wrapper).toMatchSnapshot();
  });
  it("renders correct label", () => {
    expect(mounted.find("Chip").at(0).props().label).toBe("Test label");
  });
});
