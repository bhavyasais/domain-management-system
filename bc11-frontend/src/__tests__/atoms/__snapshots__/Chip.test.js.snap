// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Chip/> Renders correctly 1`] = `
<div
  className="MuiChip-root Chip-chip-1 MuiChip-sizeSmall"
  onKeyDown={[Function]}
  onKeyUp={[Function]}
  style={
    Object {
      "backgroundColor": "#eeeeee",
      "color": "#1c1c21",
    }
  }
>
  <span
    className="MuiChip-label MuiChip-labelSmall"
  >
    Test label
  </span>
</div>
`;
